pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -147,8 +147,9 @@ func (j *RayJob) PodSets() ([]kueue.PodSet, error) {
 			Name:  submitterJobPodSetName,
 			Count: 1,
 		}
-
 		submitterJobPodSet.Template = *getSubmitterTemplate(j)
+		// submitter ObjectMeta might be only available from getSubmitterTemplate and in that case it is not present j.Spec.SubmitterPodTemplate
```

Comment:
TBH I don't understand the comment. Can you rephrase please? In particular it says "**might** be only available", isn't this deterministic? What are the conditions which determine that?

Reply:
Ok, I think I understand the change after sync with @mszadkow on slack. I proposed a rephrase in https://github.com/kubernetes-sigs/kueue/pull/4340/files#r1969235726